### PR TITLE
docker: Move Dockerfile files for "base" image from the root of the repository to docker/base.

### PR DESCRIPTION
**Pull request by michael-berlin**
This unifies the locations of most of our Docker configurations and reduces complexity.

In the past, we probably had to put the Dockerfile for the "base" image in the root for the automated build feature of Docker Hub. But now they support custom paths for the Dockerfile and we should no longer have to do this.

Since the automated build looks for a README.md in the same file as the Dockerfile I created a new one in docker/base/. For the most part it is copied from the root directory, but I removed the badges at the top and the reference to Gitter.

Additionally, I've created docker/README.md which lists our different Docker images and describes how they are used in detail.

### CODE REVIEW

File: docker/README.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+**TL;DR:** Use the [vitess/lite](https://hub.docker.com/r/vitess/lite/) image for running Vitess.
+Our Kubernetes Tutorial uses it as well.
+Instead of using the `latest` tag, you can pin it to a known stable version e.g. `v2.0`.
+
+## Principles
+
+The structure of this directory and our Dockerfile files is guided by the following principles:
+
+* The configuration of an image is usually in the directory `docker/<image>/`.
```
Comment (pivanof) [inline docker/README.md]: "usually" means there are exceptions, but there aren't any, right? Or if you meant that the configuration for k8s dependencies and tools are in different directories, then can you rephrase this to tell that? E.g. "The configurations of all Vitess images are in a directory `docker/<image>/` (configurations for Kubernetes dependencies and tools are in other places)."

File: docker/README.md [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+| **base** | automatic (after every GitHub push to the master branch) | Contains all Vitess server binaries. Snapshot after running `make build`. |
+| **root** | automatic (after every GitHub push to the master branch) | Same as **base** but with the default user set to "root". Required for Kubernetes. |
+| **lite** | manual (updated with every Vitess release) | Stripped down version of **base** e.g. source code and build dependencies are removed. Default image in our Kubernetes templates for minimized startup time. |
+
+All these Vitess images include a specific MySQL/MariaDB version ("flavor").
+
+  * We provide Dockerfile files for multiple flavors (`Dockerfile.<flavor>`).
+  * As of April 2017, the following flavors are supported: `mariadb`, `mysql56`, `mysql57`, `percona`(56), `percona57`
+  * On Docker Hub we publishe only images with MySQL 5.7 to minimize maintenance overhead and avoid confusion.
```
Comment (pivanof) [inline docker/README.md]: s/publishe/publish/

File: docker/README.md (hunk 1 above)
Comment (michael-berlin) [inline docker/README.md]: Great suggestion! I changed it.

File: docker/README.md (hunk 2 above)
Comment (michael-berlin) [inline docker/README.md]: Done.
